benchmarks/benchmarks/optimize.py

```diff
@@ -234,3 +236,42 @@ def run_LJ(self, methods=None):
         for i in range(10):
             b.bench_run(np.random.uniform(-2,2,natoms*3), methods=methods)
         return b
+
+
+class BenchLeastSquares(Benchmark):
+    """Class for benchmarking nonlinear least squares solvers."""
+    problems = extract_lsq_problems()
+    params = [
+        list(problems.keys()),
+        ["average time", "nfev", "success"]
+    ]
+    param_names = [
+        "problem", "result type"
+    ]
+
+    def track_all(self, problem_name, result_type):
+        problem = self.problems[problem_name]
+
+        if problem.lb is not None or problem.ub is not None:
+            raise NotImplementedError
+
+        ftol = 1e-5
+
+        if result_type == 'average time':
+            n_runs = 10
+            t0 = time.time()
+            for _ in range(n_runs):
+                leastsq(problem.fun, problem.x0, Dfun=problem.jac, ftol=ftol,
+                        full_output=True)
+            return (time.time() - t0) / n_runs
```

Comment:
This is probably OK, but I cannot help but notice that ASV selects the number of runs with a more sophisticated way:
https://github.com/spacetelescope/asv/blob/master/asv/benchmark.py#L451

@pv what is the recommended way of making multiple measurements on a benchmark?